src/libponyc/pass/refer.c

```diff
@@ -139,14 +139,16 @@ static const char* generate_multi_dot_name(ast_t* ast, ast_t** def_found) {
 
     default:
     {
+      if (def == NULL)
+        return stringtab("");
```

Comment:
@jemc , this is one change needed in `generate_multi_dot_name`.  
I believe this is

* will do no harm to working pony code:  The new code was added in a place where it is throwing an assert. So no working code will get affected.

* The change is in sync with the rest of the method: if the parent ast node has null `data`,  we are supposed to return empty string. That is possible when we have expressions wrapped in braces.